### PR TITLE
feat(connector): [PayU] Use connector_response_response_id as reference to merchant

### LINKED ISSUE #2344: [FEATURE]: [PayU] Use `connector_response_reference_id` as reference to merchant
### :memo: Feature Description

- Reference id are used to map transactions in the connector’s dashboard.
- Hyperswitch manages several reference ids, such as `payment_id`, `attempt_id`, and `connector_transaction_id` for a single transaction. 
- However, merchants may encounter uncertainty when determining which ID to utilize in the connector dashboard to identify the payment.

### :hammer: Possible Implementation

- When we receive a response from the connector for the payment, we deserialize it and populate the `response` field in RouterData. 
- For the `TransactionResponse` type, we must fill the `connector_response_reference_id` with a corresponding reference id for the merchant to identify the transaction. 
- One might need to have exposure to api docs of the connector for which it is being implemented to decide what to fill in connector_response_reference_id.
- You can check this PR for further reference https://github.com/juspay/hyperswitch/pull/1735

🔖 Note: All the changes needed should be contained within `hyperswitch/crates/router/src/connector/`

### :package: Have you spent some time checking if this feature request has been raised before?

- [X] I checked and didn't find a similar issue

### :package: Have you read the Contributing Guidelines?

- [X] I have read the [Contributing Guidelines](https://github.com/juspay/hyperswitch/blob/main/docs/CONTRIBUTING.md)

### PR DESCRIPTION
**Pull request by harshalkh**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description
Fixes #2344 

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## Motivation and Context
Fixes #2344 

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

File: crates/router/src/connector/payu/transformers.rs [hunk 1]
```diff
@@ -199,7 +199,9 @@ impl<F, T>
                 mandate_reference: None,
                 connector_metadata: None,
                 network_txn_id: None,
-                connector_response_reference_id: None,
+                connector_response_reference_id: Some(
+                    item.response.ext_order_id.unwrap_or(item.response.order_id)
+                ),
```
Comment (srujanchikke) [inline crates/router/src/connector/payu/transformers.rs:204]:
```suggestion
                connector_response_reference_id: item.response.ext_order_id.or(Some(item.response.order_id)),
```
Hello @harshalkh ,

You are unwrapping and then if the value is Some we are wrapping it again, whereas we can use or to reduce that one redundant operation . Please change it in other places as well .

**Comment by harshalkh** [comment]
@srujanchikke thanks for review. I have done required changes..

**Comment by harshalkh** [comment]
@srujanchikke I have done required changes, do I need to use clone as well here? Or not required?

**Comment by srujanchikke** [comment]
> @srujanchikke I have done required changes, do I need to use clone as well here? Or not required?

Yes , you need to use clone here

```
resource_id: types::ResponseId::ConnectorTransactionId(item.response.order_id.clone()),

**Comment by srujanchikke** [comment]
Hey @harshalkh ,

Please use the following cargo commands -

`cargo check --all-features`                                                                 
`cargo clippy --all-features` 

ref -[ Cargo commands](https://github.com/juspay/hyperswitch/blob/main/docs/CONTRIBUTING.md#cargo-commands), and fix the errors thrown by clippy for the checks to pass .

Please run `cargo +nightly fmt`  to fix failing formatting check

**Comment by harshalkh** [comment]
> Hey @harshalkh ,
> 
> Please use the following cargo commands -
> 
> `cargo check --all-features` `cargo clippy --all-features`
> 
> ref -[ Cargo commands](https://github.com/juspay/hyperswitch/blob/main/docs/CONTRIBUTING.md#cargo-commands), and fix the errors thrown by clippy for the checks to pass .
> 
> Please run `cargo +nightly fmt` to fix failing formatting check

Thanks I will check

**Review by srujanchikke — APPROVED** [review]
Looks good to me!

**Review by SanchithHegde — APPROVED** [review]
Looks good to me!

Thanks for the PR, @harshalkh!

**Comment by VedantKhairnar** [comment]
Hey @harshalkh  👋,
Thanks a bunch for all your contributions! We've got a little something for you to show our appreciation.
Just take a moment to fill out this [form](https://forms.gle/yqbtjFuP6oxLaBKN8), and get ready for some awesome swag coming your way.
Thanks!

**Pull request merged**